client/state/account-recovery/actions.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * External dependencies
+ */
+import wpcom from 'lib/wp';
+
+/**
+ * Internal dependencies
+ */
+import {
+	ACCOUNT_RECOVERY_FETCH,
+	ACCOUNT_RECOVERY_FETCH_SUCCESS,
+	ACCOUNT_RECOVERY_FETCH_FAILED,
+} from 'state/action-types';
+
+export const accountRecoveryFetchSuccess = ( accountRecoverySettings ) => {
+	return {
+		type: ACCOUNT_RECOVERY_FETCH_SUCCESS,
+		accountRecoverySettings,
+	};
+};
+
+export const accountRecoveryFetchFailed = ( error ) => {
+	return {
+		type: ACCOUNT_RECOVERY_FETCH_FAILED,
+		error,
+	};
+};
+
+export const accountRecoveryFetch = () => ( dispatch ) => {
+	dispatch( { type: ACCOUNT_RECOVERY_FETCH } );
+
+	return new Promise( ( resolve, reject ) => {
```

Comment:
Seeing this return here confused me a bit as actions are supposed to return a thunk or an action object but I see this is the return value for the thunk so in my opinion its a little unnecessary. Just a nitpick :)